binary_transparency/firmware/cmd/ft_personality/internal/http/server.go

```diff
@@ -85,27 +86,26 @@ func NewServer(c Trillian, cas CAS) *Server {
 func (s *Server) addFirmware(w http.ResponseWriter, r *http.Request) {
 	statement, image, err := parseAddFirmwareRequest(r)
 	if err != nil {
-		http.Error(w, err.Error(), http.StatusBadRequest)
+		http.Error(w, fmt.Sprintf("failed to parse request: %q", err.Error()), http.StatusBadRequest)
 		return
 	}
 
 	stmt := api.FirmwareStatement{}
 	if err := json.NewDecoder(bytes.NewReader(statement)).Decode(&stmt); err != nil {
-		http.Error(w, err.Error(), http.StatusBadRequest)
+		http.Error(w, fmt.Sprintf("failed to decode statement: %q", err.Error()), http.StatusBadRequest)
 		return
 	}
 
 	// "Verify" the signature:
-	// TODO(al): do proper sigs
-	if sigStr := string(stmt.Signature); sigStr != "LOL!" {
-		http.Error(w, fmt.Sprintf("invalid LOL! sig %q", sigStr), http.StatusBadRequest)
+	if ok, err := common.VerifySignature(stmt.Metadata, stmt.Signature); !ok {
+		http.Error(w, fmt.Sprintf("Signature verification failed! %v", err), http.StatusBadRequest)
```

Comment:
lower-case "Signature" for consistency with other error messages

Reply:
ACK